supertokens_python/recipe/thirdpartypasswordless/recipeimplementation/implementation.py

```diff
@@ -160,41 +161,25 @@ async def create_code(self,
                           email: Union[None, str],
                           phone_number: Union[None, str],
                           user_input_code: Union[None, str],
-                          user_context: Dict[str, Any]) -> CreateCodeResult:
+                          user_context: Dict[str, Any]) -> CreateCodeOkResult:
         return await self.pless_create_code(email, phone_number, user_input_code, user_context)
 
     async def create_new_code_for_device(self,
                                          device_id: str,
                                          user_input_code: Union[str, None],
-                                         user_context: Dict[str, Any]) -> CreateNewCodeForDeviceResult:
+                                         user_context: Dict[str, Any]) -> Union[CreateNewCodeForDeviceOkResult, CreateNewCodeForDeviceRestartFlowErrorResult, CreateNewCodeForDeviceUserInputCodeAlreadyUsedErrorResult]:
         return await self.pless_create_new_code_for_device(device_id, user_input_code, user_context)
 
     async def consume_code(self,
                            pre_auth_session_id: str,
                            user_input_code: Union[str, None],
                            device_id: Union[str, None],
                            link_code: Union[str, None],
-                           user_context: Dict[str, Any]) -> ConsumeCodeResult:
-        result = await self.pless_consume_code(pre_auth_session_id, user_input_code, device_id, link_code, user_context)
-
-        if result.is_ok:
-            if result.user is None or result.created_new_user is None:
-                raise Exception("Should never come here")
-            return ConsumeCodeOkResult(result.created_new_user, User(result.user.user_id, result.user.email, result.user.phone_number, None, result.user.time_joined))
-        if result.is_expired_user_input_code_error:
-            if result.failed_code_input_attempt_count is None or result.maximum_code_input_attempts is None:
-                raise Exception("Should never come here")
-            return ConsumeCodeExpiredUserInputCodeErrorResult(result.failed_code_input_attempt_count, result.maximum_code_input_attempts)
-        if result.is_incorrect_user_input_code_error:
-            if result.failed_code_input_attempt_count is None or result.maximum_code_input_attempts is None:
-                raise Exception("Should never come here")
-            return ConsumeCodeIncorrectUserInputCodeErrorResult(result.failed_code_input_attempt_count, result.maximum_code_input_attempts)
-
-        # restart flow error
-        return ConsumeCodeRestartFlowErrorResult()
+                           user_context: Dict[str, Any]) -> Union[ConsumeCodeOkResult, ConsumeCodeIncorrectUserInputCodeErrorResult, ConsumeCodeExpiredUserInputCodeErrorResult, ConsumeCodeRestartFlowErrorResult]:
+        return await self.pless_consume_code(pre_auth_session_id, user_input_code, device_id, link_code, user_context)
```

Comment:
This will change cause we will return an interface from thirdpartpasswordless for types that contain user.